PicPi/FrameAPI.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  FrameAPI.swift
+//  PicPi
+//
+//  Created by Mustafa T. Mohammed on 12/21/20.
+//
+
+import Foundation
+import Promises
+
+func injectFrameAPI() -> FrameAPI {
+  return FrameAPIImpl()
+}
+/**
+ # Example usage
+ 
+ ## Initialization
+   ```
+  let frameAPI: FrameAPI
+  init(frameAPI: FrameAPI = injectFrameAPI()) {
+   self.frameAPI = frameAPI
+  }
+ ```
+## Call function that return for example [URL]
+ - Notice that we use (.then) that will return the list of URLs and (.catch) that will return the rejection reason (error)
+ ```
+  func getThumbnails() {
+   frameAPI.retrieveThumbnails().then {  Urls    in
+     print(Urls)`
+   }.catch {reject in
+     print(reject)
+    }
+  }
+ ```
+ */
```

Comment:
Great example! Mind converting this to /// vs /* * /

Reply:
Thank you!
don't you think it's easier to use /** */  for multi-line docs?

Reply:
It is less type but it show up when you look up the docs on the class in a different file?

Reply:
yes it does  show up!

Reply:
Ooh! nice, okay whichever works for you sgtm